src/watertap_contrib/reflo/unit_models/tests/test_solar_still.py

```diff
@@ -0,0 +1,247 @@
+    degrees_of_freedom,
+    number_variables,
+    number_total_constraints,
+    number_unused_variables,
+)
+from idaes.core.util.scaling import (
+    calculate_scaling_factors,
+    unscaled_variables_generator,
+)
+
+from watertap.core.solvers import get_solver
+from watertap.property_models.multicomp_aq_sol_prop_pack import (
+    MCASParameterBlock,
+)
+
+from watertap_contrib.reflo.unit_models import SolarStill
+from watertap_contrib.reflo.costing import TreatmentCosting
+
+# Get default solver for testing
+solver = get_solver()
+
+__location__ = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
+test_data_path = f"{__location__}/solar_still_test_data_phoenix_az.csv"
+
+
+def build_ss():
+
+    rho = 1000 * pyunits.kg / pyunits.m**3
+    inlet_dict = {
+        "solute_list": ["TDS"],
+        "mw_data": {"TDS": 31.4038218e-3},
+        "material_flow_basis": MaterialFlowBasis.mass,
+    }
+
+    water_yield_calc_dict = dict(
+        input_weather_file_path=test_data_path,  # path to input weather file
+        interval_day=15,  # interval at which to calculate water yield (e.g., every 15 days)
+        salinity=20,  # salinity of influent water; g/L
+        water_depth_basin=0.02,  # depth of water in solar still basin; m
+        length_basin=0.6,  # length of each side of basin (length=width); m
+    )
+
+    tds_conc = 20 * pyunits.g / pyunits.liter
+    daily_water_production = 100 * pyunits.m**3 / pyunits.day
+
+    flow_mass_in = pyunits.convert(
+        daily_water_production * rho, to_units=pyunits.kg / pyunits.s
+    )
+    flow_mass_tds = pyunits.convert(
+        daily_water_production * tds_conc, to_units=pyunits.kg / pyunits.s
+    )
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(dynamic=False)
+    m.fs.properties = MCASParameterBlock(**inlet_dict)
+
+    m.fs.unit = SolarStill(
+        property_package=m.fs.properties,
+        water_yield_calculation_args=water_yield_calc_dict,
+    )
+
+    m.fs.unit.properties_in[0].flow_vol_phase[...]
+    m.fs.unit.properties_in[0].flow_mass_phase_comp["Liq", "H2O"].fix(
+        value(flow_mass_in)
+    )
+    m.fs.unit.properties_in[0].flow_mass_phase_comp["Liq", "TDS"].fix(
+        value(flow_mass_tds)
+    )
+    m.fs.unit.properties_in[0].pressure.fix(101325)
+    m.fs.unit.properties_in[0].temperature.fix(293.15)
+
+    return m
+
+
+class TestSolarStill:
+    @pytest.fixture(scope="class")
+    def ss_frame(self):
+        m = build_ss()
+        return m
+
+    @pytest.mark.unit
+    def test_config(self, ss_frame):
+        m = ss_frame
+        assert len(m.fs.unit.config) == 5
+
+        assert not m.fs.unit.config.dynamic
+        assert not m.fs.unit.config.has_holdup
+        assert m.fs.unit.config.property_package is m.fs.properties
+        assert_units_consistent(m)
+
+    @pytest.mark.unit
+    def test_build(self, ss_frame):
+        m = ss_frame
+
+        port_list = ["inlet", "outlet", "waste"]
```

Comment:
Wondering if the port naming should follow the convention for ZO models in WaterTAP (`inlet`,`treated`,`byproduct`), unless this is really a 0D model.

Reply:
I have never been a fan of `treated` or `byproduct` tbh. But you are right that I don't think this is a 0D model.